### ISSUE TITLE
Get back to original pre-refactor performance

### ISSUE THREAD

**Issue by rpeltekov** [rpeltekov/ge3t_shim_tool#12]
- [x] fix the eval scan procedure as well, so that i can queue basis check comparisons
- [x] need to make sure shim scan ALL slices works and schedules all relevant slices properly
- [x] that the histograms are being saved correctly, 
- [x] #17
- [x] make sure shim actually works effectively
        - this involves making sure the scans are actually performing optimally, as they are expected to be doing so. 
        - will likely need to increase the TE to reduce noise in the B0Map acquisitions, and then can hopefully get clearer image
        - to increase TE, will probably need to finish #5
        - if that still does not work, will need to characterize gradients better to see if they are actually linear #16, which is dependent on #2

**Comment by rpeltekov** [comment]
trying to get the second task working. stumbling here where too many slices are selected...

![image](https://github.com/rpeltekov/ge3t_shim_tool/assets/18520782/ce047332-d952-4a48-9fc8-eb03f4236613)

**Comment by rpeltekov** [comment]
it looks like to get the "doAllShimmedScans" selecting the correct slices to actually attempt shimming on, the easiest marker / sign of a valid slice with signal after ROI selection is found by whether the slice of the BackgroundB0Map contains any statistics. i.e. is shimTool.shimStats[0][i] is None.
![Screenshot from 2024-05-21 11-06-01](https://github.com/rpeltekov/ge3t_shim_tool/assets/18520782/7e4919cb-1d10-4b94-b2b1-1bf2ef85fb82)

**Comment by rpeltekov** [comment]
moving the rest to a new issue, #19 , because we have reached pre-refactor performance, passing it now with the phase unwrapping already being implemented. this "bug" has been reached, and to declutter we close it
